app/lib/rejection_reasons/builder.rb

```diff
@@ -0,0 +1,28 @@
+module RejectionReasons
+  class Builder
+    CONFIG_PATH = 'config/rejection_reasons.yml'.freeze
+
+    def rejection_reasons
+      @rejection_reasons ||= build_reasons(configuration[:reasons])
+    end
+
+  private
+
+    def configuration
+      @configuration ||= YAML.load_file(CONFIG_PATH)
+    end
+
+    def build_reasons(reasons_ary)
+      reasons_ary.map do |hash|
+        reason = RejectionReasons::Reason.new(hash)
+        reason.details = build_details(hash[:details]) if hash.key?(:details)
```

Comment:
Maybe we can have a rejections reasons main model that does the building instead of a builder? and has methods to access question objects which themselves build the details/or nested reasons rather than here?

Reply:
not sure if recursiveness hides the structure though but happy to discuss/hear your thoughts

Reply:
My thinking is that currently we have `Reason` which is always a checkbox element or a nested checkbox, and `Details` which is always a textarea element nested within a reason. The original `Question`/`Section` classes from the spike felt arbitrary, though I imagine as we drive out this feature the config will become more complex.

Reply:
@cwrw I've attempted a more implicit build pattern as mentioned, might not be quite what you had in mind, happy to discuss if you like.